src/ucp/api/ucp_def.h

```diff
@@ -612,11 +612,13 @@ typedef ucs_status_t (*ucp_am_callback_t)(void *arg, void *data, size_t length,
  * The callback is always called from the progress context, therefore calling
  * communication progress routines from this callback is not allowed. It is
  * recommended to define callbacks with relatively short execution time to
- * avoid blocking of communication progress.
+ * avoid blocking of communication progress. The callback is thread safe with
```

Comment:
I think what you are trying to say: "User provided callback must be thread safe in respect to the worker object". Can we invoke UCP call from the callback ?

Reply:
I think we have to mention that thread safety is guaranteed with exactly that worker which was used for registering this callback. Yes, we can call UCP communication call, but not progress (which is mentioned few lines above)

Reply:
1. calling communication progress routines from this callback is not allowed -> `calling @ref ucp_worker_progress() is not allowed". THere is only one communication progress routine (and we plan to keep it this way)
2. Thread safety - not sure why need special comment here - the thread safe guarantees is as with any other API. The note "The callback is always called from the progress context" is sufficient for that matter.
@shamisp WDYT?